modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -128,13 +123,13 @@ private QueryAnalyzer() {
      * @param indexVersion  The create version of the index containing the percolator queries.
      */
     static Result analyze(Query query, Version indexVersion) {
-        Class<?> queryClass = query.getClass();
+        Class<? extends Query> queryClass = query.getClass();
```

Comment:
I pushed 3cad419a385ebe1a27a3ee9fb096f4ce55f40f3b. My IDE was throwing a warning at me that the map lookup was suspicious because `queryClass` might not be `Class<? extends Query>` but I think that 3cad419a385ebe1a27a3ee9fb096f4ce55f40f3b is a better way to ensure this and remove the IDE warning.